osf/models/metadata.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+from django.db import models
+
+from addons.osfstorage.models import OsfStorageFile
+from osf.models.base import BaseModel, ObjectIDMixin
+from osf.models.metaschema import FileMetadataMetaSchema
+from osf.utils.datetime_aware_jsonfield import DateTimeAwareJSONField
+
+
+class FileMetadataRecord(ObjectIDMixin, BaseModel):
+
+    metadata = DateTimeAwareJSONField(default=dict, blank=True)
+
+    file = models.ForeignKey(OsfStorageFile, related_name='records', on_delete=models.SET_NULL, null=True)
+    schema = models.ForeignKey(FileMetadataMetaSchema, related_name='records', on_delete=models.SET_NULL, null=True)
+
+    class Meta:
+        unique_together = ('file', 'schema')
+
+    def format(self, type='json'):
+        return self.schema.formatter.format(self.file, self.metadata, type)
```

Comment:
Remove self.file and self.metadata args here. Just type

Reply:
Naming needs work. Potentially `self.schema.serializer.serialize()` (defaults to json) or `self.schema.serializer.serialize(format='xml')`